src/dsp_tools/commands/xml_validate/models/data_rdf.py

```diff
@@ -0,0 +1,211 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from rdflib import RDF
+from rdflib import RDFS
+from rdflib import BNode
+from rdflib import Graph
+from rdflib import Literal
+from rdflib import Namespace
+from rdflib import URIRef
+
+KNORA_API = Namespace("http://api.knora.org/ontology/knora-api/v2#")
+API_SHAPES = Namespace("http://api.knora.org/ontology/knora-api/shapes/v2#")
+
+
+@dataclass
+class DataRDF:
+    shortcode: str
+    default_onto: str
+    resources: list[ResourceRDF]
+
+    def make_graph(self) -> Graph:
+        g = Graph()
+        for r in self.resources:
+            g += r.make_graph()
+        return g
+
+
+@dataclass
+class ResourceRDF:
+    res_id: URIRef
+    res_class: URIRef
+    label: Literal
+    values: list[ValueRDF]
+
+    def make_graph(self) -> Graph:
+        g = Graph()
+        g.add((self.res_id, RDF.type, self.res_class))
+        g.add((self.res_id, RDFS.label, self.label))
+        for v in self.values:
+            g += v.make_graph(self.res_id)
+        return g
+
+
+@dataclass
+class ValueRDF:
```

Comment:
You could mark this as abstract base class (https://docs.python.org/3/library/abc.html#abc.ABC):

```suggestion
class ValueRDF(ABC):
```

Reply:
Mypy is not amused with that. It used to be a protocol, but I changed it. Because of that.